src/modules/analytics/sagas.ts

```diff
@@ -0,0 +1,89 @@
+import { takeLatest, select } from 'redux-saga/effects'
+import { getAnalytics } from 'decentraland-dapps/dist/modules/analytics/utils'
+
+import { OPEN_EDITOR, OpenEditorAction, TOGGLE_SNAP_TO_GRID, ToggleSnapToGridAction } from 'modules/editor/actions'
+import { getCurrentProject, getProject } from 'modules/project/selectors'
+import {
+  ADD_ITEM,
+  DUPLICATE_ITEM,
+  AddItemAction,
+  DuplicateItemAction,
+  DeleteItemAction,
+  DELETE_ITEM,
+  SET_GROUND,
+  SetGroundAction
+} from 'modules/scene/actions'
+import { SUBMIT_PROJECT_SUCCESS, SubmitProjectSuccessAction } from 'modules/contest/actions'
+import { RootState } from 'modules/common/types'
+import { getScene } from 'modules/scene/selectors'
+
+export function* segmentSaga() {
+  yield takeLatest(OPEN_EDITOR, handleOpenEditor)
+  yield takeLatest(ADD_ITEM, handleAddItem)
+  yield takeLatest(DUPLICATE_ITEM, handleDuplicateItem)
+  yield takeLatest(SET_GROUND, handleSetGround)
+  yield takeLatest(DELETE_ITEM, handleRemoveItem)
+  yield takeLatest(SUBMIT_PROJECT_SUCCESS, handleSubmitProject)
+  yield takeLatest(TOGGLE_SNAP_TO_GRID, handleToggleSnapToGrid)
+}
+
+function* handleOpenEditor(_: OpenEditorAction) {
+  const project: ReturnType<typeof getCurrentProject> = yield select(getCurrentProject)
+  if (!project) return
+
+  getAnalytics().track('Open project', { projectId: project.id })
+}
+
+function* handleAddItem(_: AddItemAction) {
+  const project: ReturnType<typeof getCurrentProject> = yield select(getCurrentProject)
+  if (!project) return
+
+  getAnalytics().track('New item', { projectId: project.id })
+}
+
+function* handleDuplicateItem(_: DuplicateItemAction) {
+  const project: ReturnType<typeof getCurrentProject> = yield select(getCurrentProject)
+  if (!project) return
+
+  getAnalytics().track('New item', { projectId: project.id })
+}
+
+function* handleSetGround(_: SetGroundAction) {
+  const project: ReturnType<typeof getCurrentProject> = yield select(getCurrentProject)
+  if (!project) return
+
+  getAnalytics().track('New item', { projectId: project.id })
+}
```

Comment:
Not sure about this one, but you could have a `handleNewItem` which handles `ADD_ITEM, DUPLICATE_ITEM and SET_GROUND`

```ts
yield takeLatest(ADD_ITEM, handleNewItem)
yield takeLatest(DUPLICATE_ITEM, handleNewItem)
yield takeLatest(SET_GROUND, handleNewItem)```

Reply:
oh yea, this was due to a refactor 😅 mala mia. I was sending also the asset, and the way to get the asset was different for each action. Then I simplified it and now the 3 actions could be handled by the same handler. I'll fix it.